src/validation.cairo

```diff
@@ -38,11 +39,98 @@ impl BlockValidatorImpl of BlockValidator {
 }
 
 #[generate_trait]
-impl TransactionValidatorImpl of TransactionValidator {
+pub impl TransactionValidatorImpl of TransactionValidator {
+    // marker, flag, and witness fields in segwit transactions are not included
+    // this means txid computation is the same for legacy and segwit tx
     fn txid(self: @Transaction) -> u256 {
-        // TODO: implement
-        0
+        let version: felt252 = (*self.version).into();
```

Comment:
I would rather apply into within append_word invocation, like instead of:
```
let version: felt252 = (*self.version).into();
```
and:
```
sha256_input.append_word(version, 1);
```
just:
``` 
sha256_input.append_word((*self.version).into(), 1);
```
To me it is easier to read.

Reply:
sure, i'll do that!